api/util/errors.go

```diff
@@ -71,3 +73,9 @@ func RespondWithError(w http.ResponseWriter, err error) {
 	WriteErrResponse(w, http.StatusInternalServerError, err)
 	return
 }
+
+// RespondWithDispatchTypeError(w http.ResponseWriter, got
+func RespondWithDispatchTypeError(w http.ResponseWriter, got interface{}) {
+	log.Errorf("type mismatch: %v of type %s", got, reflect.TypeOf(got))
+	WriteErrResponse(w, http.StatusInternalServerError, fmt.Errorf("type mismatch: %v of type %s", got, reflect.TypeOf(got)))
```

Comment:
Great to have this utility function! Just wanted to point out that this is an appropriate use of `StatusInternalServerError`, since this error should only be hit if there's a logical / programming error in qri core - some method returning the wrong type from its declared method - rather than bad user input or a losing a network connection or something like that.